packages/core/src/prompts/prompt.ts

```diff
@@ -145,8 +145,24 @@ export default class Prompt {
       this.state = 'active';
     }
 
-    if (key?.name && keys.has(key.name)) {
-      this.emit('cursor', key.name);
+    switch (key?.name) {
```

Comment:
Since this is the base prompt, which all other prompts extend, wouldn't this – at least, potentially others – break the `text` prompt?

There is no normal/insert modes a la `vi` for the text input, it would always be in insert mode.

Reply:
Nevermind, it's up to each prompt which events to subscribe to and act on. My bad.